graphql_compiler/compiler/emit_sql.py

```diff
@@ -1187,7 +1191,12 @@ def fold(self, fold_scope_location: FoldScopeLocation) -> None:
 
         # 1. Get fold metadata.
         # Location of vertex that is folded on.
-        outer_alias = self._current_alias.alias()
+        if isinstance(self._current_alias, CTE):
```

Comment:
`self._current_alias` is typed `Optional[Alias]`, but I don't think `CTE` is a type of `Alias`. Seems weird that that doesn't cause a mypy error, but perhaps the type of `self._current_alias` ought to updated?

Reply:
Based on `reveal_type`, somewhere along the line the type of `self._current_alias` is turning into `Any`-- looking into why now.